background/index.js

```diff
@@ -17,13 +17,171 @@
 /* global chrome */
 
 import Processor from './processor';
+import Ws from './ws';
 
-const processor = new Processor();
+import { UI, TRANSPORT_UNINITIALIZED } from '../shared';
 
 chrome.runtime.onConnect.addListener((port) => {
-  console.assert(port.name === 'id');
+  if (port.name === 'secureApi') {
+    port.onMessage.addListener(web3Message(port));
+    return;
+  }
+
+  if (port.name === 'barScripts') {
+    port.onMessage.addListener(loadScripts(port));
+    return;
+  }
+
+  // TODO [ToDr] Use connection without elevated privileges here!
+  if (port.name === 'web3') {
+    port.onMessage.addListener(web3Message(port));
+    return;
+  }
+
+  if (port.name === 'id') {
+    port.onMessage.addListener(processId(port));
+    return;
+  }
+
+  throw new Error(`Unrecognized port: ${port.name}`);
+});
+
+let openedTabId = null;
+let transport = null;
+// Attempt to extract token on start if not available.
+extractToken();
+
+chrome.runtime.onMessage.addListener((request, sender, callback) => {
+  if (!(transport && transport.isConnected) && request.token) {
+    if (transport) {
+      // TODO [ToDr] kill old transport!
+    }
+
+    if (openedTabId) {
+      chrome.tabs.remove(openedTabId);
```

Comment:
That tab is opened in the background (and it always opens only once), we also check if the node is up before opening it.

I think it might be better to just display a welcome page (that's actually quite a common thing for extensions) loaded from `http://127.0.0.1:8180/#/welcome` - it could list extension features, and the token would be extracted automatically :)